gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestTaskIntegrationSpec.groovy

```diff
@@ -180,6 +180,26 @@ class CreateManifestTaskIntegrationSpec extends GradleIntegrationSpec {
         buildResult.output.contains('Encountered product dependency declaration that was also ignored')
     }
 
+    def 'throws if declared dependency is also optional'() {
+        setup:
+        buildFile << """
+            createManifest {
+                productDependencies = [
```

Comment:
ditto about using the expected API here (ie `productDependency {`), even if the other tests don't do this.

Reply:
Ditto on above, it does not work directly. I believe this is the reason tests are written this way (I just copied and edited given existing templates)